packages/web-api-scan-runner/src/runner/url-deduper.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable } from 'inversify';
+import { uniq } from 'lodash';
+
+@injectable
+export class UrlDeduper {
+    public dedupe: (discoveredUrls: string[], newUrls: string[]) => string[] = (discoveredUrls, newUrls) => {
+        return uniq(discoveredUrls.concat(newUrls));
```

Comment:
Doesn't necessarily need to be in this PR, but this could be moved into the `mergeAndWrite` function of `WebsiteScanResultProvider`. We have specific merge logic for certain array fields in WebsiteScanResult.